core/src/main/java/com/zextras/carbonio/files/rest/controllers/PreviewController.java

```diff
@@ -106,112 +96,112 @@ protected void channelRead0(
       Matcher previewDocumentMatcher = Endpoints.PREVIEW_DOCUMENT.matcher((uriRequest));
       Matcher thumbnailDocumentMatcher = Endpoints.THUMBNAIL_DOCUMENT.matcher((uriRequest));
 
-      User requester = (User) context
-        .channel()
-        .attr(AttributeKey.valueOf(ContextAttribute.REQUESTER))
-        .get();
-
-      if (thumbnailImageMatcher.find() && httpRequest.method().equals(HttpMethod.GET)) {
-        thumbnailImage(context, httpRequest, thumbnailImageMatcher, requester);
-        return;
-      }
-
-      if (thumbnailPdfMatcher.find() && httpRequest.method().equals(HttpMethod.GET)) {
-        thumbnailPdf(context, httpRequest, thumbnailPdfMatcher, requester);
-        return;
-      }
-
-      if (thumbnailDocumentMatcher.find() && httpRequest.method().equals(HttpMethod.GET)) {
-        thumbnailDocument(context, httpRequest, thumbnailDocumentMatcher, requester);
-        return;
-      }
-
-      if (previewImageMatcher.find() && httpRequest.method().equals(HttpMethod.GET)) {
-        previewImage(context, httpRequest, previewImageMatcher, requester);
-        return;
-      }
-
-      if (previewPdfMatcher.find() && httpRequest.method().equals(HttpMethod.GET)) {
-        previewPdf(context, httpRequest, previewPdfMatcher, requester);
-        return;
-      }
+      // get the user with updated locale for every preview request.
+      // if this was cached so would be the locale resulting in incorrect preview if user changes
+      // language and then
+      // requests a preview. all cookie checks here are already passed so we know there is one and
+      // if
+      // user management is not down an usermyself should always be returned given the cookie.
+      HttpHeaders headersRequest = httpRequest.headers();
+      String cookiesString = headersRequest.get(HttpHeaderNames.COOKIE);
+      Optional<UserMyself> requester = userRepository.getUserMyselfByCookieNotCached(cookiesString);
+
+      if (requester.isEmpty()) {
+        logger.warn(
+            String.format("Request %s %s: unauthorized", httpRequest.method(), httpRequest.uri()));
+
+        context.fireExceptionCaught(new UnAuthorized());
```

Comment:
I would rewrite this in a more linear way. If you are at this point of the execution we already know that the requester is authorized and we already have the `User` object in the `REQUESTER` channel attribute. So, if something goes wrong during the UserMyself fetching, I would map the `User` object to a `UserMyself` with a default Locale.

I think this method could be part of the UserMyself pojo since it is the one that could have more and more attributes in the future.

```suggestion
      UserMyself requester = userRepository
        .getUserMyselfByCookieNotCached(cookiesString)
        .orElseGet(UserMyself.mapFromUser((User) context
           .channel()
           .attr(AttributeKey.valueOf(ContextAttribute.REQUESTER))
           .get()));
```